File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
  void solve(TreeNode* root, vector<int> &inorder){
    if(!root) return;
    if(root->left) solve(root->left, inorder);
    inorder.push_back(root->val);
    if(root->right) solve(root->right, inorder);
    return;
  }
    int kthSmallest(TreeNode* root, int k) {
        vector<int> inorder;
        solve(root, inorder);
        return inorder[k-1];
    }
};

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    private int ans = -1;
    private int count = 0;

    public void Inorder(TreeNode root, int k) {
        if (root == null)
            return;
        Inorder(root.left, k);
        count++;
        if (count == k) {
            ans = root.val;
            return;
        }

        Inorder(root.right, k);
    }

    public int kthSmallest(TreeNode root, int k) {
        Inorder(root, k);
        return ans;
    }
}